### PR TITLE
fixes #3821 Plugin menu's are lost after page refresh in development mod...

### PR DESCRIPTION
**Pull request by abenari**
...e

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -68,6 +68,9 @@ class Application < Rails::Application
     config.autoload_paths += %W(#{config.root}/app/models/trends)
     config.autoload_paths += %W(#{config.root}/app/models/taxonomies)
 
+    # on development environment load the following paths once
+    # this is useful to keep singletons from reloading.
+    config.autoload_once_paths += Dir["#{config.root}/app/services/foreman"]
```
Comment (domcleal) [inline config/application.rb]: Can we be more specific, or perhaps split singletons out?  I'd rather not exclude this whole dir.

**Comment by ehelms** [comment]
From my testing, I did not encounter my previous issues with the menu disappearing.

**Comment by abenari** [comment]
@domcleal I didn't find a way to specify a single file in the auto_load_once path.
Splitting the singleton code out might make the code less readable, as it will push it into a separate module and namespace, I think code clarity is more important here then the need to restart the service when code changes in this folder.

**Comment by abenari** [comment]
[test]

**Comment by abenari** [comment]
@domcleal I couldn't make the "auto load once" work with better granularity.
Since the directories have code that does not change frequently and some of it runs only on server initialization, I think that restarting the server on each change is reasonable solution.

**Comment by domcleal** [comment]
@abenari okay, I'll live with that.  However `rake test` isn't even working...

**Comment by abenari** [comment]
[test]

**Comment by domcleal** [comment]
Thanks @abenari, merged as 9ef39dd6 for Foreman 1.4.0.

**Pull request closed without merging**